test_agent/rust/src/testagent.rs

```diff
@@ -149,7 +149,7 @@ impl SocketTestAgent {
 
     async fn handle_send_command(
         &self,
-        utransport: &UTransportSocket,
+        utransport: &dyn UTransport,
```

Comment:
_Typically_ if we're passing around trait objects we'd like to wrap them in a `Box` or `Arc`. `Box` is fine for if you don't need any sort of multi-threaded access to the contained trait object in the heap. `Arc` stands for Atomic Reference Count and is what's used in a multi-threaded context.

Which do you think is a better fit?

Reply:
Did it with Box. Any methods we're calling (at least right now) will be in a single thread